test_workspace/BinNat.bosatsu

```diff
@@ -71,6 +67,10 @@ def cmp_BinNat(a: BinNat, b: BinNat) -> Comparison:
             case GT | EQ: GT
             case LT: LT
         case Zero: GT
+    case Zero:
```

Comment:
this is more efficient since Zero is rare, but triggered the bug before this fix.